modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/rate/swap/CashFlowEquivalentCalculator.java

```diff
@@ -415,6 +433,42 @@ public static ResolvedSwapLeg cashFlowEquivalentOnLeg(
     return sorted;
   }
 
+  /**
+   * Generate a new map with each payment date unique and the amounts and sensitivities of elements 
+   * with same payment date compressed.
+   * <p>
+   * The original map is unchanged.
+   * 
+   * @param input  the starting map
+   * @return the normalized map
+   */
+  public static Map<Payment, PointSensitivityBuilder> normalize(Map<Payment, PointSensitivityBuilder> input) {
```

Comment:
i've got to confess something here: i had trouble following this code. it also seems rather fragile (esp. as it's modifying the looping index inside the loop). might i recommend something along these lines (probably less efficient, but a lot clearer - at least to me):

```
public static Map<Payment, PointSensitivityBuilder> normalize(Map<Payment, PointSensitivityBuilder> input) {

    Map<Payment, PointSensitivityBuilder> output = new HashMap<>();

    // create a map of the payment(s) associated to each date
    SortedMap<LocalDate, List<Payment>> paymentMap = new TreeMap<>();
    for (Payment payment : input.keySet()) {
      paymentMap.computeIfAbsent(payment.getDate(), l -> new ArrayList<>());
      paymentMap.get(payment.getDate()).add(payment);
    }

    // compress the payments and sensitivities which share a payment date
    for (LocalDate date : paymentMap.keySet()) {
      CurrencyAmount compressedAmount = paymentMap.get(date).stream()
          .map(payment -> payment.getValue()).reduce(CurrencyAmount::plus).get();
      Payment compressedPayment = Payment.of(compressedAmount, date);

      PointSensitivityBuilder compressedSensiBuilder = paymentMap.get(date).stream()
          .map(payment -> input.get(payment))
          .reduce(PointSensitivityBuilder.none(), PointSensitivityBuilder::combinedWith);

      output.put(compressedPayment, compressedSensiBuilder);
    }
    return output;
  }
```

Reply:
ok to leave as-is as this is in concordance with existing strata approaches